### PR TITLE
[TECHNICAL] Unit tests for datasource classes - Part 1

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App:

- [ ] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    @Test
+    fun `getUrlToOpenInWeb returns an url to open in website`() {
+        val expectedUrl = "https://example.com/file123/TestApp"
+        val getUrlToOpenInWebResult: RemoteOperationResult<String> = createRemoteOperationResultMock(data = expectedUrl, isSuccess = true)
+
+        every {
+            ocAppRegistryService.getUrlToOpenInWeb(
+                openWebEndpoint = openWebEndpoint,
+                fileId = OC_FILE.remoteId.toString(),
+                appName = appName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:79]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    @Test(expected = Exception::class)
+    fun `getAppRegistryForAccount returns an Exception when getAppRegistry() has an error controlled by an Exception`() {
+        every { ocAppRegistryService.getAppRegistry() } throws Exception()
+
+        ocRemoteAppRegistryDataSource.getAppRegistryForAccount(OC_ACCOUNT_NAME)
+    }
+
+    @Test
+    fun `getUrlToOpenInWeb returns an url to open in website`() {
+        val expectedUrl = "https://example.com/file123/TestApp"
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:72]: There is no constant defined to use here? If there isn't I will create e new one.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                openWebEndpoint = openWebEndpoint,
+                fileId = OC_FILE.remoteId.toString(),
+                appName = appName
+            )
+        } returns getUrlToOpenInWebResult
+
+        val result = ocAppRegistryService.getUrlToOpenInWeb(
+            openWebEndpoint = openWebEndpoint,
+            fileId = OC_FILE.remoteId.toString(),
+            appName = appName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:86]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        )
+
+        assertEquals(expectedUrl, result.data)
+
+        verify {
+            ocAppRegistryService.getUrlToOpenInWeb(
+                openWebEndpoint = openWebEndpoint,
+                fileId = OC_FILE.remoteId.toString(),
+                appName = appName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:95]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    @Test(expected = Exception::class)
+    fun `getUrlToOpenInWeb returns an exception when something there is an error in the method`() {
+
+        every {
+            ocAppRegistryService.getUrlToOpenInWeb(
+                openWebEndpoint = openWebEndpoint,
+                fileId = OC_FILE.remoteId.toString(),
+                appName = appName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:107]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            )
+        } throws Exception()
+
+        ocAppRegistryService.getUrlToOpenInWeb(
+            openWebEndpoint = openWebEndpoint,
+            fileId = OC_FILE.remoteId.toString(),
+            appName = appName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:114]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    fun `createFileWithAppProvider returns the url to open in web`() {
+
+        val expectedFileUrl = "https://example.com/files/testFile.txt"
+        val createFileWithAppProviderResult: RemoteOperationResult<String> = createRemoteOperationResultMock(data = expectedFileUrl, isSuccess = true)
+
+        every {
+            ocAppRegistryService.createFileWithAppProvider(
+                createFileWithAppProviderEndpoint = openWebEndpoint,
+                parentContainerId = OC_FILE.remoteId.toString(),
+                filename = OC_FILE.fileName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:128]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+            openWebEndpoint = openWebEndpoint,
+            fileId = OC_FILE.remoteId.toString(),
+            appName = appName
+        )
+    }
+
+    @Test
+    fun `createFileWithAppProvider returns the url to open in web`() {
+
+        val expectedFileUrl = "https://example.com/files/testFile.txt"
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:121]: There is no constant defined to use here? If there isn't I will create e new one.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+                createFileWithAppProviderEndpoint = openWebEndpoint,
+                parentContainerId = OC_FILE.remoteId.toString(),
+                filename = OC_FILE.fileName
+            )
+        } returns createFileWithAppProviderResult
+
+        val result = ocAppRegistryService.createFileWithAppProvider(
+            createFileWithAppProviderEndpoint = openWebEndpoint,
+            parentContainerId = OC_FILE.remoteId.toString(),
+            filename = OC_FILE.fileName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:135]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+        )
+
+        assertEquals(expectedFileUrl, result.data)
+
+        verify(exactly = 1) {
+            ocAppRegistryService.createFileWithAppProvider(
+                createFileWithAppProviderEndpoint = openWebEndpoint,
+                parentContainerId = OC_FILE.remoteId.toString(),
+                filename = OC_FILE.fileName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:144]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    @Test(expected = Exception::class)
+    fun `createFileWithAppProvider returns an Exception in method`() {
+
+        every {
+            ocAppRegistryService.createFileWithAppProvider(
+                createFileWithAppProviderEndpoint = openWebEndpoint,
+                parentContainerId = OC_FILE.remoteId.toString(),
+                filename = OC_FILE.fileName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:156]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+            )
+        } throws Exception()
+
+        ocAppRegistryService.createFileWithAppProvider(
+            createFileWithAppProviderEndpoint = openWebEndpoint,
+            parentContainerId = OC_FILE.remoteId.toString(),
+            filename = OC_FILE.fileName
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:163]: A comma is missing here.

File: owncloudData/src/test/java/com/owncloud/android/data/capabilities/datasources/OCLocalCapabilitiesDataSourceTest.kt [hunk 13]
```diff
@@ -58,7 +59,7 @@ class OCLocalCapabilitiesDataSourceTest {
     }
 
     @Test
-    fun getCapabilitiesForAccountAsLiveData() {
+    fun `getCapabilitiesForAccountAsLiveData returns a livedata`() {
```
Comment (manuelplazaspalacio) [inline owncloudData/src/test/java/com/owncloud/android/data/capabilities/datasources/OCLocalCapabilitiesDataSourceTest.kt:62]: I will add more descriptive names for the tests of this class.

File: owncloudTestUtil/src/main/java/com/owncloud/android/testutil/OCAppRegistryMimeType.kt [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+
+val OC_APP_REGISTRY_MIMETYPE = AppRegistryMimeType(
+    mimeType = "DIR",
+    ext = "appRegistryMimeTypes.ext",
+    appProviders = emptyList(),
+    name = "appRegistryMimeTypes.name",
+    icon = "appRegistryMimeTypes.icon",
+    description= "appRegistryMimeTypes.description",
+    allowCreation=true,
+    defaultApplication= "appRegistryMimeTypes.defaultApplication")
```
Comment (manuelplazaspalacio) [inline owncloudTestUtil/src/main/java/com/owncloud/android/testutil/OCAppRegistryMimeType.kt:13]: A comma is missing here. And put the ) in the next line.

**Review by manuelplazaspalacio — COMMENTED** [review]
Some changes here @Aitorbp

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt (hunk 1 above)
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:79]: done

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt (hunk 2 above)
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:72]: I didn't find any.

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt (hunk 3 above)
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:86]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt (hunk 4 above)
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:95]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt (hunk 5 above)
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:107]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt (hunk 6 above)
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:114]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt (hunk 7 above)
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt:128]: Done

File: owncloudTestUtil/src/main/java/com/owncloud/android/testutil/OCAppRegistryMimeType.kt (hunk 14 above)
Comment (manuelplazaspalacio) [reply owncloudTestUtil/src/main/java/com/owncloud/android/testutil/OCAppRegistryMimeType.kt]: You forgot to put the ')' in the next line.

File: owncloudData/src/test/java/com/owncloud/android/data/capabilities/datasources/OCLocalCapabilitiesDataSourceTest.kt (hunk 13 above)
Comment (manuelplazaspalacio) [reply owncloudData/src/test/java/com/owncloud/android/data/capabilities/datasources/OCLocalCapabilitiesDataSourceTest.kt]: You didn´t add  more descriptive names for the rest of the tests.

File: owncloudTestUtil/src/main/java/com/owncloud/android/testutil/OCAppRegistryMimeType.kt (hunk 14 above)
Comment (Aitorbp) [reply owncloudTestUtil/src/main/java/com/owncloud/android/testutil/OCAppRegistryMimeType.kt]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/capabilities/datasources/OCLocalCapabilitiesDataSourceTest.kt (hunk 13 above)
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/capabilities/datasources/OCLocalCapabilitiesDataSourceTest.kt]: Done

**Review by manuelplazaspalacio — APPROVED** [review]
LGTM!!

**Pull request merged**